### PR TITLE
When using an accessor, strip const when the type contains only mutable values and const references

### PR DESCRIPTION
**Pull request by FeepingCreature**
When using an accessor, strip const when the data type contains only mutable values and const references.
In that case, all non-const data is value copied anyways.

### CODE REVIEW

File: src/accessors.d [hunk 1]
```diff
@@ -389,6 +444,17 @@ private template generateAttributeString(uint attributes)
     }
 }
 
+private enum mayGainfullyStripConst(T) = !is(T == Unqual!T) && isAssignable!(Unqual!T, T);
```
Comment (linkrope) [inline src/accessors.d]: omit "Gainfully": the check only ensures that `const` can be stripped, but not whether this is gainful

File: src/accessors.d [hunk 2]
```diff
@@ -156,15 +154,31 @@ template GenerateReader(string name, alias field)
...
-        }
         else
         {
-            return format("%s final @property auto %s() inout %s{ return this.%s; }",
-                visibility, accessorName, attributesString, name);
+            string returnStmt;
+
+            static if (mayGainfullyStripConst!(typeof(field)))
+            {
+                returnStmt = format(
```
Comment (linkrope) [inline src/accessors.d]: `returnStmt` is not the right name for such a statement list

File: src/accessors.d (hunk 1 above)
Comment (FeepingCreature) [reply src/accessors.d]: "gainfully" is checked via !is(T == Unqual!T), ie. whether T actually has any attributes to strip. If not, we can omit the stripping code for a cleaner, leaner accessor.

File: src/accessors.d [hunk 3]
```diff
@@ -244,12 +263,48 @@ template GenerateConstReader(string name, alias field)
 
         alias attributes = inferAttributes!(typeof(field), "__postblit");
         string attributesString = generateAttributeString!attributes;
+        string returnStmt;
```
Comment (linkrope) [inline src/accessors.d:266]: `accessorBody`

**Review by linkrope — CHANGES_REQUESTED** [review]

**Comment by FeepingCreature** [comment]
boilerplate is subsuming accessors. This change is already implemented there.

**Pull request closed without merging**